src/components/AssetInput/MultiAssetPreview/Assets.tsx

```diff
@@ -0,0 +1,44 @@
+import { merge } from "@utilities/merge";
+import React, { FC } from "react";
+import { AssetInputSize } from "../AssetInput";
+import { AssetThumbnail } from "../AssetThumbnail";
+import { MultiAssetPreviewProps } from "./MultiAssetPreview";
+
+export const Assets: FC<Pick<MultiAssetPreviewProps, "assets">> = ({ assets }) => {
+    const length = assets?.length || 0;
+    const assetsNew = assets?.slice(0, 4) || [];
```

Comment:
I'd maybe rename this to `previewAssets` to make it more explicit what the array is used for